pymc3/tests/test_distributions_random.py

```diff
@@ -517,6 +525,37 @@ def test_probability_vector_shape(self):
         assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 3, 7)
 
 
+@pytest.mark.skipif(polyagamma_not_installed)
+class PolyaGamma(BaseTestCases.BaseTestCase):
+    rng = np.random.default_rng(SeededTest.random_seed)
+    distribution = pg_distribution
+    params = {"h": 1, "z": 0.0}
+
+    @staticmethod
+    def sample_random_variable(this, size):
+        """ Draws samples from a RandomVariable using its .rng_fn method. """
+        if size is None:
+            return this.distribution.rv_op.rng_fn(cls.rng)
+        else:
+            return this.distribution.rv_op.rng_fn(cls.rng, size=size)
+
+    @pytest.mark.parametrize(
+        "shape, size",
+        [
+            ((2), (1)),
+            ((2), (2)),
+            ((2, 2), (2, 100)),
+            ((3, 4), (3, 4)),
+            ((3, 4), (3, 4, 100)),
+            ((3, 4), (100)),
+            ((3, 4), (1)),
+        ],
+    )
+    def test_polyagamma_random_shape(self, shape, size):
+        out_shape = to_tuple(size) + to_tuple(shape)
+        assert self.distribution.rv_op.rng_fn(cls.rng, size=size).shape == out_shape
```

Comment:
You only need to make sure that the parameter mappings from `Distribution` to `RandomVariable` to `random_polyagamma` are correct; otherwise, the `polyagamma` package is responsible for testing the correctness of its samples.  

In other words, we have to be careful not to introduce 3rd party-specific tests to this package.